labgrid/strategy/ubootstrategy.py

```diff
@@ -57,3 +57,17 @@ def transition(self, status):
         else:
             raise StrategyError(f"no transition found from {self.status} to {status}")
         self.status = status
+
+    def force(self, status):
+        if not isinstance(status, Status):
+            status = Status[status]
+        if status == Status.off:
+            self.target.activate(self.power)
+        elif status == Status.uboot:
+            self.target.activate(self.console)
```

Comment:
You can drop this line. The activation of the UBootDriver (which binds against the ConsoleProtocol) in the next line activates it anyway. Activating the ConsoleProtocol explicitly is only needed if we need to to it early, i.e. before turning power on, so we don't miss early characters on the serial interface.

Reply:
Line removed. I tested the transition form initial uboot state to shell state, it works!